### DISCUSSION TITLE
SQL IN with list of integer values

### DISCUSSION THREAD

**Discussion by frbelotto** [sqlalchemy/sqlalchemy#8377] [Usage Questions] [answered]
### Describe the bug

Hello,
I am running a SQL statements with the IN operator that receives a list of integers.

This is the SQL "pure" command :
```
select * from db2coc.ctt_cli_cmco t1 
where t1.cd_acao = 52976 and t1.CD_BLOC in (321360, 321361)
```
I am getting such data to a dataframe using pandas and this is the result :
![image](https://user-images.githubusercontent.com/24688087/183989018-640b2ff8-37e6-4e2f-a930-c3f1c63ca30a.png)

But now I try the same results with SQLA:

```
cod_acao_COC = [52976]
cod_estrato_acao_COC = [321360, 321361]
select_stmt_COC = select(contatos_COC)
select_stmt_COC = select_stmt_COC.where(contatos_COC.c.cod_acao == cod_acao_COC)
select_stmt_COC = select_stmt_COC.where(contatos_COC.c.cod_estrato.in_([cod_estrato_acao_COC]))
```

And this are the results :
![image](https://user-images.githubusercontent.com/24688087/183990119-2891a41f-8d14-49eb-90da-6bbea342639c.png)

If you take a look, you will notice that the second item in the list are not being selected (321361)

### To Reproduce

```python
This is the SQLA echo log:

2022-08-10 15:32:49,902 INFO sqlalchemy.engine.Engine BEGIN (implicit)
2022-08-10 15:32:49,903 INFO sqlalchemy.engine.Engine SELECT "DB2COC".ctt_cli_cmco.cd_acao, "DB2COC".ctt_cli_cmco."CD_BLOC" 
FROM "DB2COC".ctt_cli_cmco 
WHERE "DB2COC".ctt_cli_cmco.cd_acao = ? AND "DB2COC".ctt_cli_cmco."CD_BLOC" IN (?)
2022-08-10 15:32:49,903 INFO sqlalchemy.engine.Engine [dialect ibm_db_sa+ibm_db_sa does not support caching 0.00066s] ([52976], [321360, 321361])
2022-08-10 15:32:50,064 INFO sqlalchemy.engine.Engine COMMIT
```

### Error

There is no error message

### Versions

- OS: 10
- Python : 3.10.5
- SQLAlchemy: 1.4.39
- Database: IBM DB2 Z/OS
- DBAPI : ibm_db_sa 0.3.8

**Comment by CaselIT** [comment, marked as answer]
Hi,

Please take care about what you are passing to the sqlalchemy query,
You are doing
```py
.where(contatos_COC.c.cod_acao == [52976])
.where(contatos_COC.c.cod_estrato.in_([[321360, 321361]]))
```
I believe you instead want to do

```py
.where(contatos_COC.c.cod_acao == 52976)
.where(contatos_COC.c.cod_estrato.in_([321360, 321361]))
```

**Reply by frbelotto** [reply]
Got it! I was wrong!  Thanks man!
